lib/cpp/src/thrift/transport/TPipeServer.cpp

```diff
@@ -323,73 +344,47 @@ void TPipeServer::close() {
-  if (!SetSecurityDescriptorDacl(sd, TRUE, acl, FALSE)) {
-    auto lastError = GetLastError();
-    LocalFree(sd);
-    LocalFree(acl);
-    GlobalOutput.perror("TPipeServer::SetSecurityDescriptorDacl() GLE=", lastError);
-    throw TTransportException(TTransportException::NOT_OPEN,
-                              "SetSecurityDescriptorDacl() failed", lastError);
+bool TNamedPipeServer::createNamedPipe(const TAutoCrit& /*lockProof*/) {
+
+  PSECURITY_DESCRIPTOR psd = NULL;
+  ULONG size = 0;
+
+  if (!ConvertStringSecurityDescriptorToSecurityDescriptorA(securityDescriptor_.c_str(),
+                                                            SDDL_REVISION_1, &psd, &size)) {
+    DWORD lastError = GetLastError();
+    GlobalOutput.perror("TPipeServer::ConvertStringSecurityDescriptorToSecurityDescriptorA() GLE=",
+                        lastError);
+    throw TTransportException(
+        TTransportException::NOT_OPEN,
+        "TPipeServer::ConvertStringSecurityDescriptorToSecurityDescriptorA() failed", lastError);
   }
 
   SECURITY_ATTRIBUTES sa;
   sa.nLength = sizeof(SECURITY_ATTRIBUTES);
-  sa.lpSecurityDescriptor = sd;
+  sa.lpSecurityDescriptor = psd;
   sa.bInheritHandle = FALSE;
 
   // Create an instance of the named pipe
-  TAutoHandle hPipe(CreateNamedPipeA(pipename_.c_str(),    // pipe name
-                                     PIPE_ACCESS_DUPLEX |  // read/write access
-                                     FILE_FLAG_OVERLAPPED, // async mode
-                                     PIPE_TYPE_BYTE |      // byte type pipe
-                                     PIPE_READMODE_BYTE,   // byte read mode
-                                     maxconns_,            // max. instances
-                                     bufsize_,             // output buffer size
-                                     bufsize_,             // input buffer size
-                                     0,                    // client time-out
-                                     &sa));                // security attributes
+  TAutoHandle hPipe(CreateNamedPipeA(pipename_.c_str(),        // pipe name
+                                     PIPE_ACCESS_DUPLEX |      // read/write access
+                                         FILE_FLAG_OVERLAPPED, // async mode
+                                     PIPE_TYPE_BYTE |          // byte type pipe
+                                         PIPE_READMODE_BYTE,   // byte read mode
+                                     maxconns_,                // max. instances
+                                     bufsize_,                 // output buffer size
+                                     bufsize_,                 // input buffer size
+                                     0,                        // client time-out
+                                     &sa));                    // security attributes
 
   auto lastError = GetLastError();
-  LocalFree(sd);
-  LocalFree(acl);
-  FreeSid(everyone_sid);
+  if (psd)
+    LocalFree(psd);
```

Comment:
can we use an RAII wrapper instead?

Reply:
I feel it's a bit overkill to add a RAII considering we only free once.